lib/brewdler/dsl.rb

```diff
@@ -1,23 +1,47 @@
 module Brewdler
   class Dsl
+    class Entry
+      attr_reader :type, :name, :extra
+
+      def initialize(type, name, extra={})
+        @type = type
+        @name = name
+        @extra = extra
+      end
+    end
+
+    attr_reader :entries
+
     def initialize(input)
       @input = input
+      @entries = []
     end
 
     def process
       instance_eval(@input)
+      self
+    end
+
+    def install
+      @entries.each do |entry|
+        case entry.type
+        when :brew then Brewdler::BrewInstaller.install(entry.name, entry.extra[:options])
+        when :cask then Brewdler::CaskInstaller.install(entry.name)
+        when :repo then Brewdler::RepoInstaller.install(entry.name)
```

Comment:
Would be nice to be able to set just the class name and extra as options here so you don't need to repeat quite so much.